controllers/machine_controller.go

```diff
@@ -92,6 +122,24 @@ func (r *MachineReconciler) SetupWithManager(mgr ctrl.Manager, options controlle
 	return nil
 }
 
+func (r *MachineReconciler) reconcileRequests(a handler.MapObject) []reconcile.Request {
+	requests := []reconcile.Request{}
+	machines, err := getActiveMachinesInCluster(context.Background(), r.Client, a.Meta.GetNamespace(), a.Meta.GetName())
+	if err != nil {
+		return requests
+	}
+	for _, m := range machines {
+		r := reconcile.Request{
+			NamespacedName: types.NamespacedName{
```

Comment:
I'm not sure if you'll be able to use it (type "aliasing"), but we have `util.ObjectKey()` that returns a `client.ObjectKey` aka `types.NamespacedName` for a given object.